mesonbuild/interpreter.py

```diff
@@ -32,11 +32,91 @@
 
 import os, sys, shutil, uuid
 import re
+import stat
 
 import importlib
 
 run_depr_printed = False
 
+# The first triad is for owner permissions, the second for group permissions,
+# and the third for others (everyone else).
+# For the 1st charater:
+#  'r' means can read
+#  '-' means not allowed
+# For the 2nd charater:
+#  'w' means can write
+#  '-' means not allowed
+# For the 3rd charater:
+#  'x' means can execute
+#  's' means can execute and setuid/setgid is set (owner/group triads only)
+#  'S' means cannot execute and setuid/setgid is set (owner/group triads only)
+#  't' means can execute and sticky bit is set ("others" triads only)
+#  'T' means cannot execute and sticky bit is set ("others" triads only)
+#  '-' means none of these are allowed
+#
+# The meanings of 'rwx' perms is not obvious for directories; see:
+# https://www.hackinglinuxexposed.com/articles/20030424.html
+#
+# For information on this notation such as setuid/setgid/sticky bits, see:
+# https://en.wikipedia.org/wiki/File_system_permissions#Symbolic_notation
+symbolic_mode_regex = re.compile('(r|-)(w|-)(x|s|S|-)' # Owner mode
+                                 '(r|-)(w|-)(x|s|S|-)' # Group mode
+                                 '(r|-)(w|-)(x|t|T|-)') # Others mode
+
+def mode_symbolic_to_bits(mode_s):
+    '''
+    Does the opposite of stat.filemode(), converts strings of the form
+    'rwxr-xr-x' to st_mode enums which can be passed to os.chmod()
+    '''
+    if mode_s == None:
```

Comment:
__[flake8]__
* [E711] comparison to None should be 'if cond is None:'

<!-- Comment by [SideCI](https://sideci.com) -->